crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java

```diff
@@ -361,12 +361,16 @@ private String extractUpdatedNameFromJacksonPropertyIfPresent(Property property)
    * @return the structural schema associated with the specified property
    */
   public T internalFrom(String name, TypeRef typeRef) {
+    return internalFromImpl(name, typeRef, new LinkedList<String>());
+  }
+
+  private T internalFromImpl(String name, TypeRef typeRef, List<String> visited) {
```

Comment:
If the insertion order is not relevant (and I don't think it is in this instance), it's probably better to use a `Set` instead of a `List`. This will also simplify the lookup part which would end up being simply a call to `contains`…

Reply:
Good suggestion, switched to a `Set` for the signature and an `HashSet` for the implementation.